solvers/amigo.py

```diff
@@ -138,6 +138,8 @@ def amigo(*args, **kwargs):
 
         self.inner_var = inner_var0
         self.outer_var = outer_var0
+
+    def warm_up(self):
         if self.framework == 'numba' or self.framework == 'jax':
```

Comment:
Often better to test inclusion like this:
```suggestion
        if self.framework in ['numba', 'jax']:
```